File: src/main/scala/resonantinduction/mechanical/mech/process/crusher/NodeMechanicalPiston.java
package resonantinduction.mechanical.mech.process.crusher;

import net.minecraftforge.common.util.ForgeDirection;
import resonantinduction.mechanical.mech.MechanicalNode;

/**
 * Created by robert on 8/28/2014.
 */
public class NodeMechanicalPiston extends MechanicalNode
{
    public NodeMechanicalPiston(TileMechanicalPiston parent)
    {
        super(parent);
        maxDeltaAngle = Math.toRadians(45);
        sharePower = false;
    }

    @Override
    public boolean canConnect(ForgeDirection dir)
    {
        return dir != ((TileMechanicalPiston)getParent()).getDirection();
    }

    @Override
    protected void revolve()
    {
        ((TileMechanicalPiston)getParent()).markRevolve = true;
    }
}

File: src/main/scala/resonantinduction/mechanical/mech/process/crusher/NodeMechanicalPiston.scala
package resonantinduction.mechanical.mech.process.crusher

import net.minecraftforge.common.util.ForgeDirection
import resonantinduction.mechanical.mech.grid.NodeMechanical

/**
 * Created by robert on 8/28/2014.
 */
class NodeMechanicalPiston(parent: TileMechanicalPiston) extends NodeMechanical(parent)
{
    override def canConnect(dir: ForgeDirection): Boolean =
    {
        return dir ne (getParent.asInstanceOf[TileMechanicalPiston]).getDirection
    }

    protected def revolve
    {
        getParent.asInstanceOf[TileMechanicalPiston].markRevolve = true
    }
}